rcl/src/rcl/publisher_impl.h

```diff
@@ -25,6 +25,9 @@ struct rcl_publisher_impl_s
   rmw_qos_profile_t actual_qos;
   rcl_context_t * context;
   rmw_publisher_t * rmw_handle;
+  // Assumption: the type_support used to create the publisher outlives any
```

Comment:
After seeing this notice in places - I think it would be better to just cache the information we need - store the `rosidl_type_hash_t`. It's a little bit bigger but it requires no assumptions about lifetimes.